app/models/user.rb

```diff
@@ -1,8 +1,41 @@
 class User < ApplicationRecord
+  has_secure_password
+
   has_many :microposts, dependent: :destroy
 
-  validates :name, presence: true,
-length: {maximum: Settings.models.user.name_max_length}
-  validates :email, presence: true,
-length: {maximum: Settings.models.user.email_max_length}
+  before_save :downcase_email
+
+  validates :name,
+            presence: true,
+            length: {maximum: Settings.models.user.name_max_length}
+  validates :email,
+            presence: true,
+            uniqueness: {case_sensitive: false},
+            length: {maximum: Settings.models.user.email_max_length},
+            format: {with: Settings.models.user.valid_email_regex}
+  validate :birthday_within_last_100_years
+
+  private
+
+  def downcase_email
+    email.downcase!
+  end
+
+  def birthday_within_last_100_years
+    return if birthday.blank?
+
+    birthday_date = birthday.is_a?(Date) ? birthday : Date.parse(birthday.to_s)
+
+    return errors.add(:birthday,
+               I18n
+               .t(
+                 "activerecord.errors.models.user.attributes.birthday.future"
+               )) if birthday_date > Time.zone.today
+
+    return errors.add(:birthday,
+               I18n
+               .t(
+                  "activerecord.errors.models.user.attributes.birthday.past_100"
+                )) if birthday_date < 100.years.ago.to_date
```

Comment:
```suggestion
    return errors.add(:birthday, :past_100) if birthday_date < 100.years.ago.to_date
```
Chỗ này e đặt messages là tên key lỗi tương ứng trong I18n sẽ gọn hơn
E test thử và fix tương tự cho những chỗ khác nha
Nhớ đưa số 100 này ra thành biến const